kedro/io/data_catalog.py

```diff
@@ -162,15 +163,17 @@ def __init__(  # noqa: PLR0913
             >>>                   save_args={"index": False})
             >>> catalog = DataCatalog(datasets={'cars': cars})
         """
-        warnings.warn(
-            "`DataCatalog` has been deprecated and will be replaced by `KedroDataCatalog`, in Kedro 1.0.0. "
-            "Currently some `KedroDataCatalog` APIs have been retained for compatibility with `DataCatalog`, including "
-            "the `datasets` property and the `get_datasets`, `_get_datasets`, `add`,` list`, `add_feed_dict`, "
-            "and `shallow_copy` methods. These will be removed or replaced with updated alternatives in Kedro 1.0.0. "
-            "For more details, refer to the documentation: "
-            "https://docs.kedro.org/en/stable/data/index.html#kedrodatacatalog-experimental-feature",
-            KedroDeprecationWarning,
-        )
+        if not is_warning_suppressed():
+            warnings.warn(
+                "`DataCatalog` will be replaced by `KedroDataCatalog` in Kedro 1.0.0, "
+                "while keeping the `DataCatalog` name. Several APIs currently available, "
+                "including `datasets`, `get_datasets`, `_get_datasets`, `add`, `list`, "
+                "`add_feed_dict`, and `shallow_copy`, will be removed or replaced. "
+                "For more details, refer to the documentation: "
+                "https://docs.kedro.org/en/stable/data/index.html#kedrodatacatalog-experimental-feature",
```

Comment:
Maybe instead of mentioning `KedroDataCatalog`, we can do something like

```suggestion
                "Several `DataCatalog` APIs currently available, "
                "including `datasets`, `get_datasets`, `_get_datasets`, `add`, `list`, "
                "`add_feed_dict`, and `shallow_copy`, will be removed or replaced in Kedro 1.0.0. "
                "For more details, refer to the documentation: "
                "https://docs.kedro.org/en/stable/data/index.html#kedrodatacatalog-experimental-feature",
```

Reply:
Also, I think this should refer to a migration guide and use `/0.19.13` rather than `/stable` (to avoid linking to a disappearing page when 1.0 is released)